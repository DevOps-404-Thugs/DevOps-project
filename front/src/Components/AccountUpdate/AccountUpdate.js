import React, { useState } from 'react'
import Axios from 'axios'

function Upload() {
	const [usernameValue, setUsernameValue] = useState("")
	const [emailValue, setEmailValue] = useState("") 

	const onUsernameChange = (event) => {
		setUsernameValue(event.currentTarget.value)
	}

	const onEmailChange = (event) => {
		setEmailValue(event.currentTarget.value)
	}

	const onSubmit = (event)=>{
		event.preventDefault();
		if(!usernameValue || !emailValue){
			return alert("Please fill in all fields!")
		}
		console.log(usernameValue)
		console.log(emailValue)
		const variables = {
			username : usernameValue,
			email : emailValue
		}
		
		Axios.put(`/account`, variables, {withCredentials: true})
					.then(response => {
						console.log(response.status)
						console.log(response)
						if(response.status === 200){
                            alert("Account updated successfully!")
                            window.location.replace('/ihomie')//automatically jump to mainpage
						}else{
							alert("You meet with an error!")
						}
                    })
                    .catch(function(error){
                        console.log(error.response)
                        if(error.response.status === 400){
							alert("This account has existed!")
						}
					})
	}
	

	return(
		<div>
			<div style={{ width: '75%', margin: '3rem auto' }}>
        <form>
					<div class="form-group">
						<label>User Name</label>
						<input type="text" onChange={onUsernameChange} class="form-control" id="username-input" placeholder="Input new username here"></input>
					</div>
					<div class="form-group">
						<label>Email</label>
						<input type="text" onChange={onEmailChange} class="form-control" id="email-input" placeholder="Input new email address here"></input>
					</div>

					<button type="submit" class="btn btn-primary" onClick={onSubmit} >Submit</button>
        </form>
			</div>
		</div>
	)


}

export default Upload;