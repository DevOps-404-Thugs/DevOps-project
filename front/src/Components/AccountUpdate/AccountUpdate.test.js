import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AccountUpdate from './AccountUpdate';


describe('AccountUpdate component', () =>{
    test("match the snapshot", () =>{
        const component = renderer.create(
            <BrowserRouter>
                <AccountUpdate/>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });  
});