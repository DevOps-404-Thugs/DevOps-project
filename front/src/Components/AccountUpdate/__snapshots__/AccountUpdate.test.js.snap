// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUpdate component match the snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "margin": "3rem auto",
        "width": "75%",
      }
    }
  >
    <form>
      <div
        class="form-group"
      >
        <label>
          User Name
        </label>
        <input
          class="form-control"
          id="username-input"
          onChange={[Function]}
          placeholder="Input new username here"
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Email
        </label>
        <input
          class="form-control"
          id="email-input"
          onChange={[Function]}
          placeholder="Input new email address here"
          type="text"
        />
      </div>
      <button
        class="btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
